obsproject/obs-studio
# Add Qt UI XML validation

### Description

Adds the [official Qt schema](https://doc.qt.io/qt-5/designer-ui-file-format.html) definition to files in the UI/forms directory.

Fixes the issues found by said schema.

![image](https://user-images.githubusercontent.com/941350/129502578-6c3e66e7-32d4-4db3-b399-827465797519.png)

Adds a CI run for PRs that modify UI files to verify they follow the schema.

### Motivation and Context

While @Warchamp7 was looking into the reason behind our Qt files being mangled by Qt Creator, I found the official schema & a way to validate it. Though this doesn't catch the issue that he is still debugging, this should catch other errors sooner.

### How Has This Been Tested?

1. In VS Code, install the [XML extension by Red Hat](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml).
2. Open OBSBasic.ui
3. Notice some errors are found, fix them (see commit 2)

### Types of changes
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
